packages/ngtools/webpack/src/transformers/replace_resources.ts

```diff
@@ -18,10 +18,20 @@ export function replaceResources(
 
     const visitNode: ts.Visitor = (node: ts.Decorator) => {
       if (ts.isClassDeclaration(node)) {
-        node.decorators = ts.visitNodes(
+        const decorators = ts.visitNodes(
```

Comment:
This was originally intentional as symbol information would be lost on the TS side and cause runtime errors due to non-Angular decorators becoming disconnected from type information.

Reply:
You mean the missing return, or not using `updateClassDeclaration`?

Reply:
the `updateClassDeclaration`.   Mutating the existing node was the intentional part.

Reply:
Okay, Ill revert that part. As long as there is a `return node` to stop the recursion it should work.

Reply:
This also may have been corrected in newer version of TS.  Can you add a comment that this should be reviewed and a test case added?

Reply:
At the moment, without this change, there is a spec which fails `rebuilds on TS file changes`, which is causing `Cannot read property 'kind' of undefined`

Reply:
As discussed on slack, we still need the `updateClassDeclaration` and without this it causes some `Karma Builder watch mode` specs to fail. 

`updateClassDeclaration` is also used by the FW when replacing resources (https://github.com/angular/angular/blob/dc10355d61a98092b5f8cabc0993d737ed2e721f/packages/compiler-cli/src/transformers/inline_resources.ts#L119) so it should be safe to use. I left a comment none-the-less.